pydantic/_internal/_decorators.py

```diff
@@ -538,141 +531,155 @@ def _wrapper2(
     return _wrapper2
 
 
-GenericPlainSerializerFunctionWithoutInfo = Callable[[Any], Any]
-FieldPlainSerializerFunctionWithoutInfo = Callable[[Any, Any], Any]
-FieldWrapSerializerFunctionWithoutInfo = Callable[[Any, Any, SerializerFunctionWrapHandler], Any]
-GeneralWrapSerializerFunctionWithoutInfo = Callable[[Any, SerializerFunctionWrapHandler], Any]
-
-AnyCoreSerializer = Union[
-    FieldPlainSerializerFunction,
-    FieldWrapSerializerFunction,
-    GeneralPlainSerializerFunction,
-    GeneralWrapSerializerFunction,
-]
+def inspect_field_serializer(serializer: Callable[..., Any], mode: Literal['plain', 'wrap']) -> tuple[bool, bool]:
+    """
+    Look at a field serializer function and determine if it is a field serializer,
+    and whether it takes an info argument.
 
-AnySerializerFunction = Union[
-    GenericPlainSerializerFunctionWithoutInfo,
-    GeneralWrapSerializerFunctionWithoutInfo,
-    AnyCoreSerializer,
-]
+    An error is raised if the function has an invalid signature.
 
+    Args:
+        serializer: The serializer function to check.
+        mode: The serializer mode, either 'plain' or 'wrap'.
 
-def make_generic_serializer(
-    serializer: AnySerializerFunction, mode: Literal['plain', 'wrap'], type: Literal['field', 'general']
-) -> Any:
-    """
-    Wrap serializers to allow ignoring the `info` argument as a convenience.
+    Returns:
+        Tuple of (is_field_serializer, info_arg)
     """
-    sig = remove_params_with_defaults(signature(serializer))
-    if is_instance_method_from_sig(serializer):
-        # for the errors below to exclude self
-        sig = Signature(parameters=list(sig.parameters.values())[1:])
+    sig = signature(serializer)
```

Comment:
Why are we dropping `remove_params_with_defaults`? It's necessary to handle for example:

```python
def func(self, foo, input, info):  ...
actual = partial(func, foo=123)
serializer('x')(actual)
```

Reply:
I thought it wasn't necessary, but maybe I'm wrong - I'll add a test for `partial`, not just `partialmethod`

Reply:
https://github.com/pydantic/pydantic/blob/main/tests/test_validators.py#L1980-L2239